app/components/channel_link/channel_link.js

```diff
@@ -39,12 +50,33 @@ export default class ChannelLink extends React.PureComponent {
         return null;
     }
 
-    handlePress = () => {
-        this.props.actions.setChannelDisplayName(this.state.channel.display_name);
-        this.props.actions.handleSelectChannel(this.state.channel.id);
+    handlePress = async () => {
+        let {channel} = this.state;
 
-        if (this.props.onChannelLinkPress) {
-            this.props.onChannelLinkPress(this.state.channel);
+        if (!channel.id && channel.display_name) {
+            const {actions, currentTeamId, currentUserId} = this.props;
+            const result = await actions.joinChannel(currentUserId, currentTeamId, null, this.props.channelName);
+
+            if (result.error || !result.data || !result.data.channel) {
+                const joinFailedMessage = {
+                    id: t('mobile.join_channel.error'),
+                    defaultMessage: "We couldn't join the channel {displayName}. Please check your connection and try again.",
+                };
+                alertErrorWithFallback(this.context.intl, result.error, joinFailedMessage, channel.display_name);
+            }
+
+            if (result?.data?.channel) {
+                channel = result.data.channel;
+            }
+        }
+
+        if (channel) {
+            this.props.actions.setChannelDisplayName(this.state.channel.display_name);
```

Comment:
just a reminder to remove this after the push notifications pr goes in

Reply:
Noted, will do that.

Reply:
@enahum Done removal of `actions.setChannelDisplayName`